File: 657_Robot_Return_To_Origin.py
Python : No extra space
class Solution:
    def judgeCircle(self, moves: str) -> bool:
        if len(moves) == 0:
            return True
        x , y = 0, 0
        for element in moves:
            if element == "U":
                x += 1
            elif element == "D":
                x -= 1
            elif element == "R":
                y += 1
            elif element == "L":
                y -= 1
        if x == 0 and y == 0:
            return True
        return False


Python : Extra space
class Solution:
    def judgeCircle(self, moves: str) -> bool:
        if len(moves) == 0:
            return True
        origin = [0, 0]
        for element in moves:
            if element == "U":
                origin[0] += 1
            elif element == "D":
                origin[0] -= 1
            elif element == "R":
                origin[1] += 1
            elif element == "L":
                origin[1] -= 1
        if origin[0] == 0 and origin[1] == 0:
            return True
        return False

File: Set1_1000/657_Robot_Return_To_Origin.java
class Solution {
    public boolean judgeCircle(String moves) {
        int x = 0;
        int y = 0;
        for (char move: moves.toCharArray()) {
            if (move == 'U')
                y += 1;
            else if (move == 'D')
                y -= 1;
            else if (move == 'R')
                x += 1;
            else if (move == 'L')
                x -= 1;
        }
        if (x == 0 && y == 0)
            return true;
        return false;
    }
}
